src/app/dataset-block/metadata-block/components/event-details/builder.events.ts

```diff
@@ -40,15 +49,120 @@ type SetPollingSourceScalarSections =
     | SetPollingSourceSection.PREPROCESS
     | SetPollingSourceSection.READ;
 
-abstract class EventSectionBuilder<TEvent> {
-    public abstract buildEventSections(event: TEvent): EventSection[];
+enum SetTransformSection {
+    INPUTS = "inputs",
+    TRANSFORM = "transform",
 }
 
-class SetPollingSourceSectionBuilder extends EventSectionBuilder<SetPollingSource> {
+type SetTransformScalarSections =
+    | SetTransformSection.INPUTS
+    | SetTransformSection.TRANSFORM;
+
+export const descriptorMapper = {
+    [EventTypes.SetPollingSource]: SET_POLLING_SOURCE_DESCRIPTORS,
+    [EventTypes.SetTransform]: SET_TRANSFORM_SOURCE_DESCRIPTORS,
+};
+
+abstract class EventSectionBuilder<TEvent> {
     private readonly UNSUPPORTED_ROW_DESCRIPTOR: EventRowDescriptor = {
-        ...SET_POLLING_SOURCE_DESCRIPTORS[`SetPollingSource.UnsupportedKey`],
+        label: "",
+        tooltip: "Unsupported value",
+        presentationComponent: UnsupportedPropertyComponent,
+        separateRowForValue: false,
+        dataTestId: "unsupportedKey",
     };
+    public abstract buildEventSections(event: TEvent): EventSection[];
+    public buildEventRows(
+        event: SetPollingSource | SetTransform,
+        section: SetPollingSourceScalarSections | SetTransformScalarSections,
+    ): EventRow[] {
+        const rows: EventRow[] = [];
+        let sectionObject;
+        const allowTypenameKey = section === SetPollingSourceSection.MERGE;
+        if (event.__typename === EventTypes.SetPollingSource) {
+            sectionObject = event[section as keyof SetPollingSource];
+        } else if (event.__typename === EventTypes.SetTransform) {
+            sectionObject = event[section as keyof SetTransform];
+        }
+        if (sectionObject) {
+            Object.entries(sectionObject).forEach(([key, value]) => {
+                if (value && (key !== "__typename" || allowTypenameKey)) {
+                    rows.push(this.buildEventRow(event, section, key, value));
+                }
+            });
+        }
 
+        return rows;
+    }
+    public buildEventRow(
+        event: SetPollingSource | SetTransform,
+        section: SetPollingSourceScalarSections | SetTransformScalarSections,
+        key: string,
+        value: unknown,
+    ): EventRow {
+        let sectionObject;
+        const eventType = event.__typename;
+        if (event.__typename === EventTypes.SetPollingSource) {
+            sectionObject =
+                event[section as keyof Omit<SetPollingSource, "__typename">];
+        }
+        if (event.__typename === EventTypes.SetTransform) {
+            sectionObject =
+                event[section as keyof Omit<SetTransform, "__typename">];
+        }
```

Comment:
The extraction procedure repeats similar one in `buildEventRows`. Just pass the section object directly and avoid this